app/services/submit_reference.rb

```diff
@@ -37,14 +37,18 @@ def progress_applications
     ActiveRecord::Base.transaction do
       reference_feedback_provided!
       application_form.application_choices.awaiting_references.each do |application_choice|
-        ApplicationStateChange.new(application_choice).references_complete!
-
-        next unless application_form.candidate_has_previously_applied?
-
-        # If the candidate has previously applied, they have less of a need to
-        # edit the application. Hence, we clear out the usual 7-day edit window
-        # by resetting the `edit_by` time.
-        application_form.update!(edit_by: Time.zone.now)
+        if EndOfCycleTimetable.stop_sending_apply_1_to_providers? && application_choice.course_option_full?
+          EndOfCycle::CancelApplicationToFullCourse.new(application_choice).call
```

Comment:
@tvararu @stevehook on second thought, I think this logical check and the invocation of `CancelApplicationToFullCourse` should sit here instead https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/services/send_application_to_provider.rb#L10 , to account for the edit_by period.

WDYT?

Reply:
👍